site/src/pages/TemplatePage/TemplateVersionsPage/TemplateVersionsPage.tsx

```diff
@@ -34,9 +38,31 @@ const TemplateVersionsPage = () => {
       displayError(getErrorMessage(error, "Failed to promote version"));
     },
   });
+
+  const { mutate: archiveVersion, isLoading: isArchiving } = useMutation({
+    mutationFn: (templateVersionId: string) => {
+      return archiveTemplateVersion(templateVersionId);
+    },
+    onSuccess: async () => {
+      // The reload is unfortunate. When a version is archived, we should hide
+      // the row. I do not know an easy way to do that, so a reload makes the API call
+      // resend and now the version is omitted.
+      // TODO: Improve this to not reload the page.
+      location.reload();
+      setSelectedVersionIdToArchive(undefined);
+      displaySuccess("Version archived successfully");
+    },
+    onError: (error) => {
+      displayError(getErrorMessage(error, "Failed to archive version"));
+    },
+  });
```

Comment:
All the queries and mutations we are putting into the api/queries modules

Reply:
I just copy pasted what was above, and changed it to archive vs promote.

Reply:
Can we refactor this in another PR?

I think I need to learn this new style of pushing the mutation to an external package. I thought it would be a copy paste, but it is not.

The current code is async and works on `useState` and callbacks, but it seems the new way is to `await`?